wmr - Display On Duty Component Info

jtasonye opened the pull request:
**Changes made in PR:**
- Added API call to `RA_OnCall.ts` file to pull the On Duty RA's information
- Filled in the needed information for `On Duty` component (name, room #, contact info, photo, profile link)
- @Mrand03 or @dli505, one of you should test it to see if the information displays in your resident view. On the backend, the `Housing.Current_On_Call`, is the view that we pull from to display this information. Since my dorm (Wilson) does not have an RD (look at `RD_Info` view), the RA on duty for Wilson does not show up in the `Housing.Current_On_Call` view, so I cannot see whether it displays, but the logic to display it should be right
- This is a joint pull request with ( https://github.com/gordon-cs/gordon-360-api/pull/1103 ), so that needs to be approved before the results from this pull request can show up.

dli505 commented on src/services/residentLife/RA_OnCall.ts:
```diff
@@ -9,4 +21,7 @@ const fetchOnDutyData = async () => {
 
-export { fetchOnDutyData };
+// Fetches the information of an On Call RA from the API endpoint "Housing/ra/on-call/{hallId}"
+const fetchOnDutyRA = (hallId: string): Promise<RA[]> => http.get(`ra/on-call/${hallId}`);
```
This should be `Housing/ra/on-call/${hallId}`!

dli505 reviewed (changes requested):
Hi Jason, great work here! Just a small bug fix where I commented and we should be good to go!

jtasonye commented:
Thanks for catching that @dli505 , I made the fix on the frontend and added some things and added some things in the backend so now it displays!

<img width="1440" alt="Screenshot 2024-11-30 at 12 11 02 AM" src="https://github.com/user-attachments/assets/5fd85a2a-31cd-433e-aeed-ae8209bd65b8">

dli505 reviewed (approved):
LGTM!
